app/models/ticket_purchase.rb

```diff
@@ -58,4 +60,11 @@ def self.update_quantity(conference, quantity, ticket, user)
     purchase.quantity = quantity if quantity > 0
     purchase
   end
+
+  def pay(payment)
+    update_attributes(paid: true, payment_id: payment)
```

Comment:
I found the mistake actually one to many association between ticket purchase and payment wasn't set up. There was no `belongs_to:payment` in ticket purchase model.